prog/dftb+/lib_extlibs/negf_int.F90

```diff
@@ -41,61 +41,53 @@ module negf_int
   use dftbp_mpifx
 #:endif
   implicit none
-  private
-
-  type(TNegf), target, public :: negf
-  ! Workaround: ifort 17, ifort 16
-  ! Passing negf for pointer dummy arguments fails despite target attribute, so pointer is needed
-  type(TNegf), pointer :: pNegf
-
-  !> general library initializations
-  public :: negf_init
 
-  !> passing structure parameters
-  public :: negf_init_str
-
-  !> initialisation of dephasing effects
-  public :: negf_init_dephasing
+  private
 
-  !> electron-phonon coupling initialisation
-  public :: negf_init_elph
+  public :: TNegfInt, TNegfInt_init, TNegfInt_final
 
-  !> clean up CSR matrices and the NEGF structure
-  public :: negf_destroy
 
-  !> wrapped functions passing dftb matrices. Needed for parallel
-  public :: calcdensity_green
+  !> Contains data needed by the NEGF interface
+  type :: TNegfInt
+    private
 
-  !> Calculate the energy weighted density from the GF
-  public :: calcEdensity_green
+    !> NEGF data container
+    type(TNegf) :: negf
 
-  !> Calculate the total current
-  public :: calc_current
+    !> compressed sparse row hamiltonian
+    type(z_CSR) :: csrHam
+    type(Z_CSR), pointer :: pCsrHam => null()
 
-  !> calculate local currents
-  public :: local_currents
+    !> compressed sparse row overlap
+    type(z_CSR) :: csrOver
 
-  !> interface csr matrices. The pattering must be predefined using negf_init_csr
-  public :: negf_init_csr
+  contains
 
-  !> compressed sparse row hamiltonian
-  type(z_CSR), target :: csrHam
-  type(Z_CSR), pointer :: pCsrHam => null()
+    procedure :: setup_csr
+    procedure :: setup_str
+    procedure :: setup_dephasing
+    procedure :: calcdensity_green
+    procedure :: calcEdensity_green
+    procedure :: calc_current
+    procedure :: local_currents
 
-  !> compressed sparse row overlap
-  type(z_CSR), target :: csrOver
-  type(Z_CSR), pointer :: pCsrOver => null()
+  end type TNegfInt
 
-  !> non wrapped direct calls
-  private :: negf_density, negf_current, negf_ldos
+  #:if WITH_MPI
+    !> Initialize MPI groups according to libNEGF needs
+    public :: negf_setup_mpi_communicators
```

Comment:
Will need to become a procedure, or have its functionality migrated into mpienv.F90

Reply:
The wrapper is indeed unnecessary, I've removed it.